### PR TITLE
fix(TextField): clearable prop not showing the clear icon

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #(issue_number)

### CODE REVIEW

File: src/components/forms/text-area/RuiTextArea.vue [hunk 1]
```diff
@@ -255,9 +255,10 @@ onMounted(computeFieldHeight);
           variant="text"
           type="button"
           icon
-          class="!p-2"
-          :color="color"
-          @click="clearIconClicked()"
+          class="!p-2 clear-btn"
```
Comment (kelsos) [inline src/components/forms/text-area/RuiTextArea.vue:258]: Wouldn't be better to split this change to it's own commit for the change log?

File: src/components/overlays/menu/RuiMenu.vue [hunk 2]
```diff
@@ -145,7 +145,7 @@ const { hasError, hasSuccess } = useFormTextDetail(
 </script>
 
 <template>
-  <div @keydown.esc="onLeave()">
+  <div @keydown.esc.stop="onLeave()">
```
Comment (kelsos) [inline src/components/overlays/menu/RuiMenu.vue:148]: Also here it's own commit for the changelog since it's a different component.

File: src/components/forms/text-field/RuiTextField.vue [hunk 3]
```diff
@@ -123,9 +122,17 @@ const showClearIcon = logicAnd(
...
   emit('clear');
 }
+
+watchDebounced(
+  focused,
+  (focused) => {
+    set(focusedDebounced, focused);
+  },
+  { debounce: 500 },
+);
```
Comment (kelsos) [inline src/components/forms/text-field/RuiTextField.vue:135]: can't we use [refDebounced](https://vueuse.org/shared/refDebounced/) from vueuse instead of a watcher here?

Comment (lukicenturi) [reply src/components/forms/text-field/RuiTextField.vue:135]: hmm yeah I should use refDebounced instead

**Review by kelsos — APPROVED** [review]
lgtm
